### DISCUSSION TITLE
Multiple queries at ones / Inner join's ?

### DISCUSSION THREAD

**Discussion by VaibhavAcharya** [supabase/supabase#886] [Questions] [answered]
Consider, I have these 3 tables :
```
users { id, name }

posts {
  id,
  text
}

likes {
  user_id,
  post_id,
  primary_key(user_id, post_id)
}
```

Now on the frontend I need to fetch some posts and also their likes.

Right now I am doing it something like this :
```
let { data: posts } = await sb.from('posts').select('*');
posts.forEach((post) => {
  const { count } = await sb.from('likes').select('*', { count: 'exact' });
  return {
    ...post,
    likes_count: count
  }
})
```
The problem here is that for every post I have to send another request just to fetch likes.

Is there a better way to do this? Can we query multiple tables at ones?

**Comment by silentworks** [comment, marked as answer]
You have two options, you can either create a view and use the `rpc` [api](https://supabase.io/docs/client/rpc) to query that view or you can write your query like:

```js
let { data: posts } = await sb.from('posts').select(`
  *,
  likes(
    *, users(*)
  )
`);
```
This should work if all your relationship keys are set in your database.

**Comment by steve-chavez** [comment]
In addition to the rpc and view options, If you only want to get post + likes count in one request, you can do it like:

```js
let { data: posts } = await sb
  .from('posts')
  .select('*,likes(count)');
```
